pandas/core/frame.py

```diff
@@ -469,6 +479,33 @@
 1   foo      8
 2   bar      7
 3   bar      8
+
+>>> df1 = pd.DataFrame({"A": [1, 2, 3], "C": [5, 6, 7]})
+>>> df2 = pd. DataFrame({"B": [1, 2, 4], "C": [7, 8, 9]})
+>>> df1
+   A  C
+0  1  5
+1  2  6
+2  3  7
+>>> df2
+   B  C
+0  1  7
+1  2  8
+2  4  9
+>>> df1.merge(df2, on="C", how="anti_left")
+   A  C   B
+0  1  5 NaN
```

Comment:
Is it common to still pull the columns from the right table in an anti join? These columns will just always be NA right?

Reply:
Thanks @WillAyd for the review.
That is true, but I am not sure. I had taken the example from a comment in the bug report.